django_prices/templatetags/prices_wrapped.py

```diff
@@ -0,0 +1,52 @@
+from decimal import Decimal, InvalidOperation
+import re
+
+from babel.core import Locale
+from babel import localedata
+from babel.numbers import format_currency
+from django import template
+from django.conf import settings
+from django.utils.safestring import mark_safe
+from django.utils.translation import get_language, to_locale
+
+register = template.Library()
+
+
+def format_price(value, currency):
+    """
+    Format decimal value as currency
+    """
+    try:
+        value = Decimal(value)
+    except (TypeError, InvalidOperation):
+        return ''
+    language = get_language()
+    if not language:
+        language = settings.LANGUAGE_CODE
+    locale_code = to_locale(language)
+    if not localedata.exists(locale_code):
+        localedata.load(locale_code)
+    locale = Locale(locale_code)
+    currency_format = locale.currency_formats.get(None)
+    pattern = currency_format.pattern
+    pattern = re.sub(
+        '(\xa4+)', '<span class="currency">\\1</span>', pattern)
+    pattern = re.sub(
+        '(#,##0.00)', '<span class="amount">\\1</span>', pattern)
```

Comment:
I tested it for various combinatons of locale and currency and it looks like it's working as expected.